### PR TITLE
etcdctl: show responses of auth RPCs if --write-output=fields is passed

### PR DESCRIPTION
**Pull request by mitake**
/cc @idonin @heyitsanthony

### CODE REVIEW

File: etcdctl/ctlv3/command/printer_fields.go [hunk 1]
```diff
@@ -146,10 +146,25 @@ func (p *fieldsPrinter) DBStatus(r dbstatus) {
...
 }
 
-func (p *fieldsPrinter) RoleAdd(role string, r v3.AuthRoleAddResponse)       { p.hdr(r.Header) }
-func (p *fieldsPrinter) RoleGet(role string, r v3.AuthRoleGetResponse)       { p.hdr(r.Header) }
+func (p *fieldsPrinter) RoleAdd(role string, r v3.AuthRoleAddResponse) { p.hdr(r.Header) }
+func (p *fieldsPrinter) RoleGet(role string, r v3.AuthRoleGetResponse) {
+	p.hdr(r.Header)
+	for _, p := range r.Perm {
+		fmt.Println(`"PermType" : `, p.PermType.String())
+		fmt.Println(`"Key" : `, string(p.Key))
```
Comment (heyitsanthony) [inline etcdctl/ctlv3/command/printer_fields.go]: ```fmt.Printf(`"Key" : %q\n`, string(p.Key))``` to match kv format / so non-printable characters display nicely

File: etcdctl/ctlv3/command/printer_fields.go [hunk 2]
```diff
@@ -146,10 +146,25 @@ func (p *fieldsPrinter) DBStatus(r dbstatus) {
...
+		fmt.Println(`"RangeEnd" : `, string(p.RangeEnd))
```
Comment (heyitsanthony) [inline etcdctl/ctlv3/command/printer_fields.go]: ```fmt.Printf(`"RangeEnd" : %q\n`, string(p.RangeEnd))```

File: etcdctl/ctlv3/command/printer_fields.go [hunk 3]
```diff
@@ -146,10 +146,25 @@ func (p *fieldsPrinter) DBStatus(r dbstatus) {
...
+		fmt.Println()
+	}
+}
 func (p *fieldsPrinter) RoleDelete(role string, r v3.AuthRoleDeleteResponse) { p.hdr(r.Header) }
-func (p *fieldsPrinter) RoleList(r v3.AuthRoleListResponse)                  { p.hdr(r.Header) }
+func (p *fieldsPrinter) RoleList(r v3.AuthRoleListResponse) {
+	p.hdr(r.Header)
+	fmt.Printf("\"Roles: \"")
```
Comment (heyitsanthony) [inline etcdctl/ctlv3/command/printer_fields.go]: ```fmt.Printf(`"Roles":`)```

File: etcdctl/ctlv3/command/printer_fields.go [hunk 4]
```diff
@@ -146,10 +146,25 @@ func (p *fieldsPrinter) DBStatus(r dbstatus) {
...
+	for _, r := range r.Roles {
+		fmt.Printf(" %s", r)
```
Comment (heyitsanthony) [inline etcdctl/ctlv3/command/printer_fields.go]: %q

**Comment by mitake** [comment]
@heyitsanthony thanks, I followed your comments. Could you take a look?

File: etcdctl/ctlv3/command/printer_fields.go [hunk 5]
```diff
@@ -146,10 +146,24 @@ func (p *fieldsPrinter) DBStatus(r dbstatus) {
...
+		fmt.Println(`"PermType" : `, p.PermType.String())
+		fmt.Printf("\"Key\" : %q\n", string(p.Key))
+		fmt.Printf("\"RangeEnd\" : %q\n", string(p.RangeEnd))
+	}
+}
 func (p *fieldsPrinter) RoleDelete(role string, r v3.AuthRoleDeleteResponse) { p.hdr(r.Header) }
-func (p *fieldsPrinter) RoleList(r v3.AuthRoleListResponse)                  { p.hdr(r.Header) }
+func (p *fieldsPrinter) RoleList(r v3.AuthRoleListResponse) {
+	p.hdr(r.Header)
+	fmt.Printf(`"Roles": `)
```
Comment (heyitsanthony) [inline etcdctl/ctlv3/command/printer_fields.go]: the spacing is still off compared to the formatting for the other field printer functions-- \`"Roles" :\`

**Comment by heyitsanthony** [comment]
lgtm after fixing the spacing. Thanks!

File: etcdctl/ctlv3/command/printer_fields.go (hunk 5 above)
Comment (mitake) [inline etcdctl/ctlv3/command/printer_fields.go]: Ah sorry, I'll fix it soon.

**Comment by mitake** [comment]
@heyitsanthony the fail of jenkins-proxy-ci seems to be fixed in https://github.com/coreos/etcd/pull/7744, so I'm merging it. Thanks!

**Pull request merged**